### PR TITLE
docs: improve readme

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

Makes the readme file more straightforward. Also fixes the error in CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       fail-fast: false
       matrix:
         platform:
-          - ubuntu-20.04
+          - ubuntu-latest
```
Comment (leslie-tsang) [inline .github/workflows/ci.yml]: No need to update this. :)
BTW, it seems nothing to do with the PR title.

Comment (pottekkat) [reply .github/workflows/ci.yml]: Reverted back.

**Comment by pottekkat** [comment]
Can we merge this?

**Comment by pottekkat** [comment]
@leslie-tsang @membphis @bzp2010 Can we merge this? I have referenced this repo in articles that have already been published. It would make sense to improve the readme so that it can be more useful.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Comment by juststillthinking** [comment]
hi @pottekkat, it looks like the ci failed, could you just update the doc, and rollback the makefile?

**Comment by pottekkat** [comment]
@monkeyDluffy6017 It still fails. Maybe we can look at this separately.

**Pull request merged**